### DISCUSSION TITLE
Functions Framework Adapter

### DISCUSSION THREAD

**Discussion by tomdavidson** [axe-me/vite-plugin-node#76] [Q&A]
What would a Custom Adapter  function look like to use [@google-cloud/functions-framework](https://github.com/GoogleCloudPlatform/functions-framework-nodejs) as a VitePluginNode adapter?  There is the http handler and the cloudevents handler.

Ignoring the cloudevent handler, the http handler has  takes a function name,  and  (res, req) => any.  In this non-functional code, `exportName` in `GCFHTTPHandler` is clearly a string and not a function.... Im just not sure what to do next.

Perhaps there is a away to access the function for VitePluginNodeConfig.exportName inside the custom adapter?

```typescript
export const GCFHTTPHandler = (exportName: string): RequestAdapter<GCFHTTP> => ({ app, req, res, server, next }) =>
  app("default", exportName(req, res));

export default defineConfig({
  server: {
    port: 3000,
  },
  plugins: [
    ...VitePluginNode({
      adapter: GCFHTTPHandler(dispatcher),
      appPath: "./src/index.ts",
      exportName: "dispatcher",
      tsCompiler: "swc",
    }),
  ],
```

**Comment by axe-me** [comment]
google cloud functions is expressjs compatible.  you can just use the express adapter. 
see the example: https://github.com/axe-me/vite-plugin-node/blob/main/examples/cloudfunction/vite.config.ts

**Reply by tomdavidson** [reply]
The Functions Framework has a different API than Express, even though it does use Express.
